index.js

```diff
@@ -72,6 +72,9 @@ class DHT extends EventEmitter {
   }
 
   static bootstrapper (port, host, opts) {
+    if (typeof host === 'object') [opts, host] = [host, undefined]
+    if (!host) host = '0.0.0.0'
```

Comment:
Host is required and needs to be your public host, docs reflect this also but let me know if something is missing there.

Reply:
Is it valid to set the host as `127.0.0.1`? As it's an unfirewalled IP
There are tests trying to use `127.0.0.1` but the host still reflects `0.0.0.0`

Reply:
What is the effect of passing the id `const id = peer.id(host, port)` in `bootstrapper()`?

Here it doesn't seem to be needed:
https://github.com/hyperswarm/testnet/blob/42bb96edcff77b1848e9d5fc1e43fea28ae280cb/index.js#L11

Anyway, it doesn't seem to fix the `0.0.0.0` host problem

Reply:
It should be the public ip of the node - ie the address it has from the outside. Only set it to 127.0.0.1 if it's a localhost only testnet. 0.0.0.0 is not a valid address, and the dev MUST provide it.

Reply:
the id is used to generate the id of the node. normally nodes figure out their own id by talking to other nodes, but bootstrappers can't since ... they are bootstrappers - so they don't immediately talk to other nodes

Reply:
We should update the code you link to use the bootstrap helper as well :)

Reply:
Understood. The `host` is just for the peer id, nothing to do with socket listening to that host lol

Reply:
And what if I want to listen on a specific address? i.e. someone might have an unfirewalled server where would like to listen on `127.0.0.1` to not expose any port

Reply:
Well, that could be no related to this PR

Reply:
Also an edge case feature for a dht, but could be a constructor option